### ISSUE TITLE
unsupported operation: more than one element of the written-to tensor refers to a single memory location

### ISSUE THREAD

**Issue by h6197627** [pytorch/pytorch#33812]
## Bug report
In recent version of PyTorch this code
```
import numpy as np
import torch
numpy_matrix = np.empty((128, 128, 3))
numpy_matrix = numpy_matrix.transpose((2, 0, 1))[np.newaxis]
torch_tensor = torch.from_numpy(numpy_matrix).cuda(device=0)
```
throws

> RuntimeError: unsupported operation: more than one element of the written-to tensor refers to a single memory location. Please clone() the tensor before performing the operation.

while for older versions it was legal code. Is it expected new behavior or some bug?

cc @ezyang @gchanan @zou3519

**Comment by ngimel** [comment]
It is a bug, thanks for reporting.

**Comment by ngimel** [comment]
The bug is not numpy-specific, removing numpy label. It happens because the stride for the 0-th dimension of size 1 is set to 0, and memory overlap check does not ignore size-1 dimensions when checking for 0 strides.

**Comment by alexjc** [comment]
I'm getting the same error running this repository with ``--cpu`` as a flag:
https://github.com/ClovaAI/WCT2

The error is in loading the pretrained modules.  I tested and it works with Pytorch 1.2.0.

**Comment by ngimel** [comment]
Bumping priority based on activity in comments

**Comment by DanielZ98** [comment]
> It is a bug, thanks for reporting.

how to solve it？

**Comment by ngimel** [comment]
It has been fixed in #34100
